### ISSUE TITLE
Intellisense error in implicit conversions in case of multiple inheritance

### ISSUE THREAD

**Issue by zajo** [microsoft/vscode-cpptools#1797]
I have:

```
class module_impl: public std::enable_shared_from_this<module_impl>, public module { .... };
```

Later on:

```
std::shared_ptr<module> create_module(....) {
  return std::make_shared<module_impl>(....);
}
```

Intellisense incorrectly reports an error at the `return` line, I'm guessing because it doesn't understand that `class module_impl` derives from `class module`:

```
no suitable user-defined conversion from "std::__1::shared_ptr<module_impl>" to "std::__1::shared_ptr<module>" exists
```

The syntax highlighting also seems wrong, notice that the second `public` doesn't light up:

![image](https://user-images.githubusercontent.com/609498/38448929-99828348-39bd-11e8-9a98-7b464acc5d0c.png)

Code Version 1.21.1 (1.21.1)
79b44aa704ce542d8ca4a3cc44cfca566e7720f1
2018-03-14T14:34:33.071Z

vscode-cpptools version 0.16.1

**Comment by bobbrow** [comment]
I can reproduce your issue in Visual Studio 2017, and we use their IntelliSense engine, so I will send this bug to them.

The syntax highlighting issue may be related to https://github.com/atom/language-c/issues/81.  I would recommend adding your case there.

**Comment by sean-mcmanus** [comment]
I think the error is a duplicate of https://github.com/Microsoft/vscode-cpptools/issues/1395, which is expected to be fixed with 0.17.8 (the fix isn't in the Insiders).

**Comment by sean-mcmanus** [comment]
This bug should have been fixed a while ago. If you still repro it, let us know if you have a way for us to repro it.
